src/core/Expander/ExpanderTitleButton/ExpanderTitleButton.test.tsx

```diff
@@ -0,0 +1,201 @@
+    expect(getByText('click to close expander')).toBeTruthy();
+  });
+});
+
+describe('Custom id', () => {
+  const TestExpanderWithProps = (props?: ExpanderTitleButtonProps) => (
+    <ExpanderTitleButton {...props}>
+      {props?.children ? props.children : 'Expander title button'}
+    </ExpanderTitleButton>
+  );
+
+  it('is passed on to button', () => {
+    const { getByRole } = customRender(TestExpanderWithProps(), {
+      providerProps: {
+        ...providerProps,
+        titleId: 'test-id_title',
+        contentId: 'test-id_content',
+      },
+    });
+    const button = getByRole('button');
+    expect(button).toHaveAttribute('id', 'test-id_title');
+    expect(button).toHaveAttribute('aria-controls', 'test-id_content');
+  });
+});
+
+describe('Provider open property', () => {
+  const TestExpanderWithProps = (props?: ExpanderTitleButtonProps) => (
+    <ExpanderTitleButton
+      {...{ 'data-testid': 'expander-open-by-default-title' }}
+      {...props}
+    >
+      {props?.children ? props.children : 'Test expander open by default'}
+    </ExpanderTitleButton>
+  );
+
+  it('correlates with aria-expanded attribute', () => {
+    const { getByRole, rerender } = customRender(TestExpanderWithProps(), {
+      providerProps: { ...providerProps, open: true },
+    });
+    const button = getByRole('button');
+    expect(button).toHaveAttribute('aria-expanded', 'true');
+    rerender(TestExpanderWithProps(), {
+      providerProps: {
+        ...providerProps,
+        open: false,
+      },
+    });
+    expect(button).toHaveAttribute('aria-expanded', 'false');
+  });
+
+  it('gives the classname to expander title and icon', () => {
+    const { getByTestId } = customRender(TestExpanderWithProps(), {
+      providerProps: { ...providerProps, open: true },
+    });
+    const div = getByTestId('expander-open-by-default-title');
+    expect(div).toHaveClass('fi-expander_title-button--open');
+    expect(div.querySelector('svg')).toHaveClass(
+      'fi-expander_title-button-icon--open',
+    );
+  });
+
+  it('will remove open classnames when false', () => {
+    const mockClickHandler = jest.fn();
+    const { getByTestId, getByRole, rerender } = customRender(
+      TestExpanderWithProps(),
+      {
+        providerProps: {
+          ...providerProps,
+          open: true,
+          onToggleExpander: mockClickHandler,
+        },
+      },
+    );
+    const buttonToClick = getByRole('button');
+    fireEvent.click(buttonToClick);
+    expect(mockClickHandler).toHaveBeenCalledTimes(1);
+    rerender(TestExpanderWithProps(), {
+      providerProps: {
+        ...providerProps,
+        open: false,
+      },
+    });
+    const titleDiv = getByTestId('expander-open-by-default-title');
+    expect(titleDiv).toHaveClass('fi-expander_title-button');
+    expect(titleDiv).not.toHaveClass('fi-expander_title-button--open');
+    expect(buttonToClick.querySelector('svg')).not.toHaveClass(
+      'fi-expander_title-button-icon--open',
```

Comment:
In general, should we have `--closed` kind of styles? E.g `fi-expander_title-button--closed`
Then we do not have to rely on `not.toHaveClass` and then it is also explicitly told.

Reply:
That's one way to do it. Would be a change to current pattern which is used in multiple places. Having it that way would be more verbose though and make things a bit more readable. Wouldn't change it just for the tests since expects are virtually free - setting up the tests seems to take most of the time consumed.

Reply:
Not just for the tests, but if it would also help otherwise for users to give custom style based on that 🤔 .
And as `not.toHaveClass` are so easily mistaken as discussed earlier, therefore could also help there.

Reply:
I wouldn't change this logic for the time being. Maybe there is a need for further discussion with the team if we need to change the overall pattern for this.